### PR TITLE
OCD-4379 - Add ROLE_ONC, ROLE_DEVELOPER, ROLE_ACB

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/CHPLHttpSecurityConfig.java [hunk 1]
```diff
@@ -93,13 +91,7 @@ public JwtSecurityConfig() {
...
 
         @Autowired
-        private JWTUserConverter userConverter;
-
-        @Autowired
-        private CognitoJwtUserConverter cognitoUserConverter;
-
-        @Autowired
-        private FF4j ff4j;
+        private JWTUserConverterFacade userConverterFactory;
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/CHPLHttpSecurityConfig.java]:
```suggestion
        private JWTUserConverterFacade userConverterFacade;
```

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/CreateOneTimeTriggerActionPermissions.java [hunk 2]
```diff
@@ -16,17 +21,27 @@ public boolean hasAccess() {
 
     @Override
     public boolean hasAccess(Object obj) {
+        LOGGER.info("ResourcePermissions type {}", getResourcePermissions().getClass().toString());
+        LOGGER.info("Object is of type {}", obj.getClass());
+        if (AuthUtil.getCurrentUser() != null) {
+            LOGGER.info("User Roles - {}", AuthUtil.getCurrentUser().getAuthorities().stream().map(ga -> ga.toString()).collect(Collectors.joining(", ")));
+        }
         if (getResourcePermissions().isUserRoleAdmin()) {
+            LOGGER.info("User is Admin");
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/CreateOneTimeTriggerActionPermissions.java]: some debug code here

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/util/AuthUtil.java [hunk 3]
```diff
@@ -3,36 +3,38 @@
...
-import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.domain.auth.Authority;
 
 public class AuthUtil {
     public static String getUsername() {
         Authentication auth = SecurityContextHolder.getContext().getAuthentication();
 
-        if (auth.getPrincipal() instanceof UserDetails) {
-            return ((UserDetails) auth.getPrincipal()).getUsername();
+        //if (auth.getPrincipal() instanceof UserDetails) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/util/AuthUtil.java]: remove commented-out code

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/util/CognitoAuthUtil.java [hunk 4]
```diff
@@ -7,8 +7,8 @@
 import org.springframework.security.core.userdetails.UserDetails;
 
 import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
-import gov.healthit.chpl.auth.user.SystemUsers;
-import gov.healthit.chpl.auth.user.User;
+import gov.healthit.chpl.auth.user.ChplSystemUsers;
+import gov.healthit.chpl.auth.user.JWTAuthenticatedUser;
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/util/CognitoAuthUtil.java:10]: checkstyle - duplicate import

**Review by kekey1 — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/CreateOneTimeTriggerActionPermissions.java [hunk 5]
```diff
@@ -16,6 +19,8 @@ public boolean hasAccess() {
 
     @Override
     public boolean hasAccess(Object obj) {
+        if (AuthUtil.getCurrentUser() != null) {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/CreateOneTimeTriggerActionPermissions.java]: Oh, checkstyle called this out - an "if" with nothing inside. Is this correct?

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/auth/authentication/JWTUserConverterFacade.java [hunk 6]
```diff
@@ -51,6 +51,10 @@ public JWTAuthenticatedUser getAuthenticatedUser(String jwt) {
 
     @Override
     public JWTAuthenticatedUser getImpersonatingUser(String jwt) throws JWTValidationException {
-        throw new NotImplementedException("CognitoJwtUserConverter.getImpersonatingUser() has not been implemented.");
+        if (ff4j.check(FeatureList.SSO)) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/auth/authentication/JWTUserConverterFacade.java]: Does this mean we cannot impersonate, at all, if the SSO flag is on? I know we're not turning it on in PROD, but I thought we were leaving it on in DEV/QA/STG for burn-in / testing purposes. If that's the case, we've basically turned off impersonation

**Review by andlar — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by andlar — APPROVED** [review]

**Review by kekey1 — APPROVED** [review]

**Pull request merged**